core/arch/arm32/sta/sub.mk

```diff
@@ -1,2 +1,7 @@
 srcs-y += sta_self_tests.c
 srcs-y += core_self_tests.c
+
+ifeq ($(CFG_SE_API),y)
+srcs-${CFG_SE_API_SELF_TEST} += se_api_self_tests.c
```

Comment:
$(CFG_SE_API_SELF_TEST) instead of ${CFG_SE_API_SELF_TEST} for consistency.
Also the ifeq is useless if you make sure CFG_SE_API_SELF_TEST is never set when CFG_SE_API is not set. Consistency between variables had better not be dealt with in sub.mk I think. So if you really want to make sure CFG_SE_API_SELF_TEST cannot be enabled whithout CFG_SE_API being also enabled, you had better do it in conf.mk (but I don't think it's really useful). That is why I would just remove the ifeq.